### DISCUSSION TITLE
Shiny error saat menggunakan shinyauthr(DB)

### DISCUSSION THREAD

**Discussion by rafiff23** [teamalgoritma/community#614] [Q&A] [answered]
https://github.com/teamalgoritma/community/discussions/608 di diskusi ini saya sudah melakukan seperti yang ada di artikel dengan menggunakan DB

code saya seperti berikut :
```
credentials <- data.frame(
  user = c("user", "Adtree"),
  password = c("user", "270792"),
  # password will automatically be hashed
  admin = c(FALSE, TRUE),
  stringsAsFactors = FALSE
)

# you can use keyring package to set database key
library(keyring)
key_set("R-shinymanager-key", "obiwankenobi")

# Init the database
create_db(
  credentials_data = credentials,
  sqlite_path = "www/database.sqlite", # will be created
  passphrase = key_get("R-shinymanager-key", "obiwankenobi")
  # passphrase = "passphrase_wihtout_keyring"
)

# dan untuk server nya
res_auth <- secure_server(
    check_credentials = check_credentials(
      "www/database.sqlite",
      passphrase = key_get("R-shinymanager-key", "obiwankenobi")
      # passphrase = "passphrase_wihtout_keyring"
    )
  )
```

saya sudah mengikuti persis seperti yang ada di artikel hanya saja saya mengubah path sql nya masuk ke folder www saya, dan saat shiny sudah di upload logs error nya seperti ini
![image](https://user-images.githubusercontent.com/90491715/230046077-d29dff34-ac0d-408d-82cf-a0a5ba6f8588.png)

**Comment by Yusraf002** [comment, marked as answer]
Dear @rafiff23  ,
Terima kasih atas pertanyaannya pak, tim kami akan segera coba bantu atau carikan referensi ya pak

**Reply by Yusraf002** [reply]
Dear @rafiff23 ,

```{r}
Error in b_file_set(self, private, service, username, keyring, propmt): 
aborted setting keyring key
calls: local ... tryCatch -> tryCatchList -> tryCatchOne -> <Anonymous>
Execution halted
```

berdasarkan error yang Bapak temukan, terindikasi bahwa ada masalah pada keyring saat mencoba mengatur passphrase. pada kode yg sudah bapa tuliskan mungkin bisa dicoba 
1. menghilangkan tanda  pagar pada  `# pada passphrase = "passphrase_without_keyring"`
2. dan memberikan tanda # pada `passphrase = key_get("R-shinymanager-key", "obiwankenobi")`


Sekian informasi dari kami, jika dirasa membantu dapat klik "mark as answer" pada comment ini. Terima kasih

Best Regards,
Algoritma Team.

**Reply by rafiff23** [reply]
terimakasih mas
